jobs/multi-arch-pipeline.Jenkinsfile

```diff
@@ -290,20 +291,42 @@ stages:
   require_artifacts: [ostree]
   post_commands:
     - cosa buildextend-qemu
-    - cosa kola run --basic-qemu-scenarios --output-dir tmp/kola-basic
-    - cosa kola run --parallel 4 --output-dir tmp/kola
+    - cosa kola run --no-test-exit-error --basic-qemu-scenarios --output-dir tmp/kola-basic
+    - cosa kola run --no-test-exit-error --parallel 4 --output-dir tmp/kola
     - cosa buildextend-metal
     - cosa buildextend-metal4k
     - cosa buildextend-live
-    - kola testiso -S --output-dir tmp/kola-metal
-    - kola testiso -SP --qemu-native-4k --scenarios iso-install --output-dir tmp/kola-metal4k
+    - kola testiso --no-test-exit-error -S --output-dir tmp/kola-metal
+    - kola testiso --no-test-exit-error -SP --qemu-native-4k --scenarios iso-install --output-dir tmp/kola-metal4k
     - cosa buildextend-aws
     - cosa buildextend-openstack
     - cosa compress --compressor xz
+    - tar --xz -cf tmp/kola.tar.xz tmp/kola-basic tmp/kola tmp/kola-metal4k tmp/kola-metal
 delay_meta_merge: false
 EOF
                    """)
             gp.gangplankArchWrapper([spec: "spec.spec", arch: basearch])
+            archiveArtifacts "builds/${newBuildID}/${basearch}/logs/kola.tar.xz"
+
+            // Check the results of the kola tests.
+            def kolatmpdir = shwrapCapture("mktemp --dir --tmpdir=\$PWD/tmp")
+            shwrap("tar xf builds/${newBuildID}/${basearch}/logs/kola.tar.xz -C ${kolatmpdir}")
+            for (koladir in ["tmp/kola-basic", "tmp/kola",
+                             "tmp/kola-metal", "tmp/kola-metal4k"]) {
+                if (!pipeutils.checkKolaSuccess("${kolatmpdir}/${koladir}", currentBuild)) {
+                    kolaSuccess = false
+                }
+            }
+            // Now that the logs are archived and checked, remove them from the
+            // builds dir and from the tmpdir we created.
+            shwrap("""
+            rm -rf ${kolatmpdir}/
+            rm -f builds/${newBuildID}/${basearch}/logs/kola.tar.xz
+            rmdir builds/${newBuildID}/${basearch}/logs
+            """)
```

Comment:
we only want to delete the logs.. the rest of the contents of `builds/${newBuildID}/${basearch}` gets uploaded to s3 later. I was trying to be careful and not use `rm -rf` just in case I messed it up somehow. 

It could be: `shwrap("rm -rf ${kolatmpdir} builds/${newBuildID}/${basearch}/logs")`

Reply:
Sorry, yup I meant `shwrap("rm -rf ${kolatmpdir} builds/${newBuildID}/${basearch}/logs")`.

Reply:
that's where it was chosen to put the files. The problem is that there's only a limited number of s3 "buckets" exposed over minio and it does make sense to keep the logs separated from other concurrent builds/tests going on. We just happen to not want them uploaded to s3 (real s3) when we upload all of the artifacts.